app/ui/ui.go

```diff
@@ -1448,6 +1466,27 @@ func (s *Server) settings(w http.ResponseWriter, r *http.Request) error {
 		return fmt.Errorf("failed to save settings: %w", err)
 	}
 
+	// Handle auto-update toggle changes
+	if old.AutoUpdateEnabled != settings.AutoUpdateEnabled {
+		if !settings.AutoUpdateEnabled {
+			// Auto-update disabled: cancel any ongoing download and clear tray notification
+			if s.Updater != nil {
+				s.Updater.CancelOngoingDownload()
+			}
+			if s.ClearUpdateAvailableFunc != nil {
+				s.ClearUpdateAvailableFunc()
```

Comment:
I believe this is the only place where we trigger the removal of the update available menu items.  I thought the logic only updated the menu with the "restart to update" after download completes.  If that's not the case, can we delay/defer the menu update so we can avoid the complexity of having to removing the menu items?